scripts/preinstall/darwin.bash.sh

```diff
@@ -1,10 +1,63 @@
 #!/bin/bash 
 
-tmp_dir=$(mktemp -d -t parsr-install) 
-
-/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" &&
-brew install qpdf imagemagick tesseract tesseract-lang tcl-tk ghostscript mupdf-tools pandoc &&
-curl https://bootstrap.pypa.io/get-pip.py -o $tmp_dir/get-pip.py &&
-python $tmp_dir/get-pip.py &&
-pip install pdfminer.six camelot-py[cv] PyPDF2 &&
-rm -rf $tmp_dir
\ No newline at end of file
+ERROR='\033[0;31m'
+OK='\033[32m'
+INFO='\033[34m'
+NO_COLOR='\033[0m'
+
+check_brew() {
+    echo -e "\n${INFO}=> Checking brew installation...${NO_COLOR}"
+    which -s brew
+    if [[ $? != 0 ]] ; then
+        echo -e "Brew is not installed. Installing..."
+        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+    else
+        brew update
+    fi
+}
+
+check_python3() {
+    echo -e "\n${INFO}=> Checking python3 installation...${NO_COLOR}"
+    which python3
+    if [[ $? != 0 ]] ; then
+        while true; do
+            read -p "Python 3 is not installed. Do you want to install it via homebrew (y/n)? " yn
+            case $yn in
+                [Yy]* ) brew install python; break;;
+                [Nn]* ) echo -e "${ERROR}Please install python3 and run the script again${NO_COLOR}"; exit 1;;
+                * ) echo "Please answer yes or no.";;
+            esac
+        done
+    else
+        echo "Python 3 is installed. Updating..."
+        brew upgrade python
+    fi
+}
+
+check_pip3() {
+    echo -e "\n${INFO}=> Checking pip3 installation...${NO_COLOR}"
+    which pip3
+    if [[ $? != 0 ]] ; then
+        echo -e "Pip 3 is not installed. Installing..."
+        python3 get-pip.py    
+    fi
+}
+
+# Check that brew, python3 and pip3 are installed
+check_brew
+check_python3
+check_pip3
+
+# Install brew dependencies
+echo -e "\n${INFO}=> Installing dependencies via brew...${NO_COLOR}"
+brew install node qpdf imagemagick tesseract tesseract-lang tcl-tk ghostscript pandoc
```

Comment:
we shouldn't install node with brew, as it can have conflicts if it was previously installed with nvm. 
Also, Node is required to run this script!

Reply:
yep, copy paste mistake. It had already removed it...